### PR TITLE
changed column selection: table `transaction_record`

### PR DESCRIPTION
**Pull request by neurosis69**
Follow up to #10300 

Basic comparison, table has only 1 row
```
sqlite> explain select transaction_record from transaction_record;
addr  opcode         p1    p2    p3    p4             p5  comment      
----  -------------  ----  ----  ----  -------------  --  -------------
0     Init           0     7     0                    00  Start at 7   
1     OpenRead       0     10    0     1              00  root=10 iDb=0; transaction_record
2     Rewind         0     6     0                    00               
3       Column         0     0     1                    00  r[1]=transaction_record.transaction_record
4       ResultRow      1     1     0                    00  output=r[1]  
5     Next           0     3     0                    01               
6     Halt           0     0     0                    00               
7     Transaction    0     0     34    0              01  usesStmtJournal=0
8     Goto           0     1     0                    00               
Run Time: real 0.001 user 0.000223 sys 0.000000
```

```
sqlite> explain select * from transaction_record;
addr  opcode         p1    p2    p3    p4             p5  comment      
----  -------------  ----  ----  ----  -------------  --  -------------
0     Init           0     18    0                    00  Start at 18  
1     OpenRead       0     10    0     12             00  root=10 iDb=0; transaction_record
2     Rewind         0     17    0                    00               
3       Column         0     0     1                    00  r[1]=transaction_record.transaction_record
4       Column         0     1     2                    00  r[2]=transaction_record.bundle_id
5       Column         0     2     3                    00  r[3]=transaction_record.confirmed_at_height
6       Column         0     3     4                    00  r[4]=transaction_record.created_at_time
7       Column         0     4     5                    00  r[5]=transaction_record.to_puzzle_hash
8       Column         0     5     6                    00  r[6]=transaction_record.amount
9       Column         0     6     7                    00  r[7]=transaction_record.fee_amount
10      Column         0     7     8                    00  r[8]=transaction_record.confirmed
11      Column         0     8     9                    00  r[9]=transaction_record.sent
12      Column         0     9     10                   00  r[10]=transaction_record.wallet_id
13      Column         0     10    11                   00  r[11]=transaction_record.trade_id
14      Column         0     11    12                   00  r[12]=transaction_record.type
15      ResultRow      1     12    0                    00  output=r[1..12]
16    Next           0     3     0                    01               
17    Halt           0     0     0                    00               
18    Transaction    0     0     34    0              01  usesStmtJournal=0
19    Goto           0     1     0                    00               
Run Time: real 0.000 user 0.000410 sys 0.000000
```

### CODE REVIEW

File: chia/wallet/wallet_transaction_store.py [hunk 1]
```diff
@@ -410,7 +415,7 @@ async def get_transactions_between(
 
     async def get_transaction_count_for_wallet(self, wallet_id) -> int:
         cursor = await self.db_connection.execute(
-            "SELECT COUNT(*) FROM transaction_record where wallet_id=?", (wallet_id,)
+            "SELECT COUNT(*) FROM transaction_record WHERE wallet_id=?", (wallet_id,)
```
Comment (arvidn) [inline chia/wallet/wallet_transaction_store.py:418]: I imagine the `*` here isn't really relevant, right? this is counting rows, not columns

**Review by arvidn — DISMISSED** [review]

File: chia/wallet/wallet_transaction_store.py (hunk 1 above)
Comment (neurosis69) [reply chia/wallet/wallet_transaction_store.py:418]: I usually use count(1) as many others do, as it is faster to type on many keyboard layouts.
but it leads to the same results.

**Comment by 1-justinstarr** [comment]
null? = nothing

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Review by emlowe — APPROVED** [review]
Looks good to me. Only fetch what you actually need from the DB seems generally a good philosophy

**Pull request merged**